gammapy/scripts/image_pipe.py

```diff
@@ -175,13 +222,102 @@ def make_maps(self, radius, bkg_norm=True):
             Disk radius in pixels for the significance map.
         bkg_norm : bool
             If true, apply the scaling factor to the bkg map
+        spectral_index : float
+            Assumed power-law spectral index
+        exposure_weighted_spectrum : bool
+            True if you want that the total excess / exposure gives the integrated flux
         """
         self.make_total_counts()
         self.make_total_bkg(bkg_norm)
         self.make_significance(radius)
+        self.make_total_excess()
+        self.make_total_exposure(spectral_index, exposure_weighted_spectrum)
 
     def make_psf(self):
         log.info('Making PSF ...')
 
-    def make_exposure(self):
-        log.info('Making EXPOSURE ...')
+    def make_total_excess(self):
+        """Compute excess between counts and bkg map."""
+        self.maps["total_excess"] = SkyMap.empty_like(self.empty_image)
+        excess = self.maps["total_counts"].data - self.maps["total_bkg"].data
+        self.maps["total_excess"].data = excess
+
+    def make_exposure(self, obs_id, spectral_index=2.3, exposure_weighted_spectrum=False):
+        """Compute the exposure map for one observation.
+
+        Excess/exposure will give the differential flux at the energy Eref at the middle of the self.energy_band
```

Comment:
No because otherwise whatever the energy_band you are calculating the same flux when you do total_excess/Exposure whereas you want the differential flux at the center of the self.energy range of the map not at 1 TeV.
Le 17 avr. 2016 à 21:48, Christoph Deil <notifications@github.com> a écrit :

> In gammapy/scripts/image_pipe.py:
> 
> >  
> >      def make_psf(self):
> >          log.info('Making PSF ...')
> >  
> > -    def make_exposure(self):
> > -        log.info('Making EXPOSURE ...')
> > +    def make_total_excess(self):
> > +        """Compute excess between counts and bkg map."""
> > +        self.maps["total_excess"] = SkyMap.empty_like(self.empty_image)
> > +        excess = self.maps["total_counts"].data - self.maps["total_bkg"].data
> > +        self.maps["total_excess"].data = excess
> > +
> > +    def make_exposure(self, obs_id, spectral_index=2.3, exposure_weighted_spectrum=False):
> > +        """Compute the exposure map for one observation.
> > +
> > +        Excess/exposure will give the differential flux at the energy Eref at the middle of the self.energy_band
> In HAP-HD and HGPS, we always use energy_ref = 1 TeV and flux_ref = 1 cm^-2 s^-1 TeV^-1.
> Can you please do the same here?
> I think that will be simpler / less error-prone for users than to compute energy_ref inside this method.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>